### PR TITLE
feat: multiple fixtures

### LINKED ISSUE #6: Add the possibility to have multiple fixtures
Hello 👋,

If i'm well understanding this package, there is only one way to configure the fixture (using the `nitro.rootDir`) in the `vitest.config.ts` but could be super useful to have the possibility to have multiple fixtures (like https://nuxt.com/docs/getting-started/testing#setup-1)

### PR DESCRIPTION
**Pull request by Barbapapazes**
resolves #6 

Hello 👋,

This tries to resolve #6 by moving the server build before all test in order to be able to configure them in the describe to allow multiple fixtures.

This works fine but:

- ~I have to disable the force trigger when the Nitro server build, otherwise, it will trigger itself.~ Because the test file will build the server, it is possible to listen for source file change to trigger a rerun.
- A setup must be defined in the `describe` bloc.

### CODE REVIEW

**Comment by johannschopplich** [comment]
Interesting! I think this is a good trede-off.

**Comment by johannschopplich** [comment]
Your draft looks well done so far. 👏 

Idea: To keep integration simple, what about automatically generating the `forceRerunTriggers` glob pattern for Nitro's source files? This could simplify the setup for newcomers while also keeping migration easy. What do you think?

**Comment by Barbapapazes** [comment]
> Your draft looks well done so far. 👏 
> 
> Idea: To keep integration simple, what about automatically generating the `forceRerunTriggers` glob pattern for Nitro's source files? This could simplify the setup for newcomers while also keeping migration easy. What do you think?

From the Nitro source directory? Yes, seems to be a really good idea!

**Comment by johannschopplich** [comment]
Yeah, I mean something like `${rootDir}/**/*.ts`. Dunno if that makes sense tho, haven't read your changes fully. 🙂

File: src/config.ts [hunk 1]
```diff
@@ -38,27 +37,11 @@ export function defineConfig(config: ViteUserConfig = {}): ViteUserConfig {
...
-        ),
-      ],
-      globalSetup: [
-        join(currentDir, 'setup.mjs'),
-      ],
-      // @ts-expect-error: Append Nitro config to access in global setup file
-      nitro,
+      forceRerunTriggers: nitro?.forceRerunTriggersOnSrcDir
+        ? [
+        `${(await loadNitroOptions()).srcDir}/**/*.ts`,
```
Comment (johannschopplich) [inline src/config.ts:42]: I think it is safer to only rerun tests when the final dev/prod server bundle has been rebuilt. Don't you think?

```ts
join(
  nitro?.rootDir || '',
  NITRO_OUTPUT_DIR,
  nitro?.mode === 'production' ? 'server' : '.nitro/dev',
  'index.mjs',
)
```

Then, we could drop the `forceRerunTriggersOnSrcDir` option altogether.

File: src/config.ts [hunk 2]
```diff
@@ -1,14 +1,15 @@
-import { fileURLToPath } from 'node:url'
-import { join } from 'pathe'
-import { defu } from 'defu'
 import { defineConfig as defineVitestConfig } from 'vitest/config'
 import { type UserConfig as ViteUserConfig, mergeConfig } from 'vite'
-import { NITRO_OUTPUT_DIR } from './constants'
+import defu from 'defu'
+import { loadOptions as loadNitroOptions } from 'nitropack'
 
 export interface NitroInlineConfig {
```
Comment (johannschopplich) [inline src/config.ts:6]: Can we bring back `mode?: 'development' | 'production'`? I want to let the users test the dev vs. prod Node bundle.

**Review by johannschopplich — COMMENTED** [review]
Great job! Two notes. 🙂

File: src/config.ts (hunk 1 above)
Comment (Barbapapazes) [reply src/config.ts:42]: > only rerun tests when the final dev/prod server bundle

The problem with that solution is:

1. Tests starts
2. Server is built => `index.mjs` is changed (something is pushed to a queue for later)
3. Tests run
4. The queue is read and this trigger a server change => back to step 2

This behavior will loop over and over again.

Also, I think it's important to allow dev to remove this behavior (addition of forceReruntrigger) if needed.

Comment (johannschopplich) [reply src/config.ts:42]: Good catch! Thanks for the explanation.

**Comment by Barbapapazes** [comment]
I think the `import.meta.dev` should be set by Nitro like Nuxt: https://github.com/nuxt/nuxt/blob/76f3ab400d0073eee31f2db68b66f1fb4709c701/packages/schema/src/config/vite.ts#L30

**Comment by johannschopplich** [comment]
Thanks for the changes! Regarding [refactor: enable prod and dev mode](https://github.com/johannschopplich/nitro-test-utils/pull/7/commits/c6fec5c41267e4571747a01d318f67467102552a), what's your reason to use `execa`? I think that is too complicated. I like the approach to just call the listener, just like before your changes. This also keeps the changes close to what Nitor tests use internally: https://github.com/unjs/nitro/blob/f2ea9f215e26782e14ddf262cfa7d544d10e88e7/test/tests.ts#L145, which opens the possibility to use the test utils later for Nitro itself.

> Can we bring back `mode?: 'development' | 'production'`? I want to let the users test the dev vs. prod Node bundle.
What I meant with that is to set `node` vs. `nitro-dev`. I don't think we need execa here. Could you please revert the changes? Thanks 🙏

**Comment by Barbapapazes** [comment]
Hey!

I revert the `execa` change. The dev can change the preset from the setup function.

**Comment by Barbapapazes** [comment]
> This also keeps the changes close to what Nitor tests use internally: [unjs/nitro@f2ea9f2/test/tests.ts#L145](https://github.com/unjs/nitro/blob/f2ea9f215e26782e14ddf262cfa7d544d10e88e7/test/tests.ts#L145), which opens the possibility to use the test utils later for Nitro itself.

Ohhh yes, I was not aware of that

**Comment by johannschopplich** [comment]
Thanks a lot. 🫠 I will merge later this week. 🙂

**Comment by johannschopplich** [comment]
Made some refactors and simplifications. I don't think it makes sense to let the user decide the Nitro preset, since only the Node build works in Vitest anyway. I think abstracting the logic provides better DX (for now).

Thanks a lot for the PR! Will update the docs later and then release this. 🚀

**Pull request merged**